src/library/searchqueryparser.cpp

```diff
@@ -114,51 +115,46 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             continue;
         }
 
+        bool negate = token.startsWith(kNegatePrefix);
+        std::unique_ptr<QueryNode> pNode;
+
         if (m_fuzzyMatcher.indexIn(token) != -1) {
             // TODO(XXX): implement this feature.
         } else if (m_textFilterMatcher.indexIn(token) != -1) {
-            bool negate = token.startsWith(kNegatePrefix);
             QString field = m_textFilterMatcher.cap(1);
             QString argument = getTextArgument(
                 m_textFilterMatcher.cap(2), &tokens).trimmed();
 
             if (!argument.isEmpty()) {
-                std::unique_ptr<QueryNode> pNode(std::make_unique<TextFilterNode>(
-                    m_database, m_fieldToSqlColumns[field], argument));
-                if (negate) {
-                    pNode = std::make_unique<NotNode>(std::move(pNode));
+                if (field == "crate") {
+                    pNode = std::make_unique<CrateFilterNode>(
+                          argument, &m_pTrackCollection->crates());
```

Comment:
`argument` is the last... argument... for the constructors of the other QueryNodes. I think the CrateFilterNode should be consistent with that pattern.